receiver/cloudfoundryreceiver/receiver_test.go

```diff
@@ -62,3 +69,651 @@ func TestDefaultValidLogsReceiver(t *testing.T) {
 	err = receiver.Shutdown(ctx)
 	require.NoError(t, err)
 }
+
+func TestSetupMetricsScope(t *testing.T) {
+	resourceSetup := pmetric.NewResourceMetrics()
+	scope := resourceSetup.ScopeMetrics().AppendEmpty()
+	scope.Scope().SetName(metadata.ScopeName)
+
+	tests := []struct {
+		name string
+		in   pmetric.ResourceMetrics
+		out  pmetric.ResourceMetrics
+	}{
+		{
+			name: "scope empty",
+			in:   pmetric.NewResourceMetrics(),
+			out:  resourceSetup,
+		},
+		{
+			name: "scope set",
+			in:   resourceSetup,
+			out:  resourceSetup,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			setupMetricsScope(tt.in)
+			require.Equal(t, tt.out, tt.in)
+		})
+	}
+}
+
+func TestGetResourceMetrics(t *testing.T) {
+	metrics := pmetric.NewMetrics()
+	resource := metrics.ResourceMetrics().AppendEmpty()
+	resource.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep")
+
+	resource2 := pmetric.NewResourceMetrics()
+	resource2.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep2")
+
+	tests := []struct {
+		name          string
+		resourceAttrs bool
+		envelope      loggregator_v2.Envelope
+		metrics       pmetric.Metrics
+		expected      pmetric.ResourceMetrics
+	}{
+		{
+			name:          "resource attributes true",
```

Comment:
It seems the name does not mack the flag

Reply:
adapted